tasks/puppet/src/main/java/com/walmartlabs/concord/plugins/puppet/PuppetClient.java

```diff
@@ -185,23 +194,26 @@ private static <T> T withRetry(int retryCount, long retryInterval, Callable<T> c
                 // probably due to self-signed cert that isn't trusted
                 log.error("Error during SSL handshake. Likely due to untrusted self-signed certificate.");
                 throw e;
+            } catch (IllegalArgumentException e) {
+                // probably invalid url
+                throw e;
             } catch (Exception e) {
                 exception = e;
                 log.error("call error", e);
             }
 
-            if (tryCount < retryCount) {
-                // take a break
-                log.info("retry after {} sec", retryInterval / 1000);
-                Utils.sleep(1000);
+            // take a break
+            log.info("retry after {} sec", retryInterval / 1000);
+            Utils.sleep(1000);
```

Comment:
Should it be `Utils.sleep(retryInterval)` instead?

Reply:
yep